lib/simple-dom/document/node.js

```diff
@@ -29,13 +32,15 @@ Node.prototype.cloneNode = function(deep) {
       child = nextChild;
     }
   }
+  node.children = getChildren(node);
 
   return node;
 };
 
 Node.prototype.appendChild = function(node) {
   if (node.nodeType === Node.DOCUMENT_FRAGMENT_NODE) {
     insertFragment(node, this, this.lastChild, null);
+    this.children = getChildren(this);
```

Comment:
This is an expensive operation we want to try and avoid.  The way ChildNodes is implemented it never loops through and collects all of the children, instead to get a particular childNode you use `el.childNodes.item(2)`, we should do the same with children. The only difference is that children excludes TextNodes.